Detectors/TPC/reconstruction/macro/getTPCTransformationExample.C

```diff
@@ -0,0 +1,100 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/*
+  Load the macro:
+  gSystem->Load("libO2TPCFastTransformation");
+  gSystem->Load("libTPCReconstruction");
+  .x getTPCTransformationExample.C
+*/
+
+#include "TPCReconstruction/TPCFastTransformHelperO2.h"
+#include "TH1F.h"
+
+using namespace o2;
+using namespace TPC;
+using namespace ali_tpc_common;
+using namespace tpc_fast_transformation;
+
+
+void spaceChargeCorrection( const double XYZ[3], double dXdYdZ[3] )
+{
+  dXdYdZ[0] = 1.;
+  dXdYdZ[1] = 2.;
+  dXdYdZ[2] = 3.;
+} 
+
+void getTPCTransformationExample()
+{
+
+  TPCFastTransformHelperO2::instance()->setSpaceChargeCorrection( spaceChargeCorrection );
+
+  std::unique_ptr<TPCFastTransform> fastTransform( TPCFastTransformHelperO2::instance()->create(0) );  
+
+  TH1F *hist = new TH1F("h","h",100,-1.e-4,1.e-4);
+
+  double statDiff=0., statN=0.;
+
+  for ( int slice = 0; slice < fastTransform->getNumberOfSlices(); slice+=1 ) {
+    std::cout<<"slice "<<slice<<" ... "<<std::endl;
+
+    const TPCFastTransform::SliceInfo &sliceInfo = fastTransform->getSliceInfo( slice );    
+
+    for ( int row = 0; row < fastTransform->getNumberOfRows(); row++ ) {
+    
+      int nPads = fastTransform->getRowInfo(row).maxPad + 1;      
+      
+      for (int pad = 0; pad < nPads; pad+=10 ) {	     
+	
+	for( float time=0; time<1000; time+=30 ){	
+	  
+	  fastTransform->setApplyDistortionFlag(0);
+	  float x0, y0, z0;
+	  int err0 = fastTransform->Transform( slice, row, pad, time, x0, y0, z0 );
+	  
+	  fastTransform->setApplyDistortionFlag(1);
+	  float x1, y1, z1;
+	  int err1 = fastTransform->Transform( slice, row, pad, time, x1, y1, z1 );
+	  
+	  if( err0!=0 || err1!=0 ){
+	    std::cout<<"can not transform!!"<<std::endl;
+	    continue;
+	  }
+
+	  // local 2 global
+	  
+	  float x0g = x0*sliceInfo.cosAlpha - y0*sliceInfo.sinAlpha;
+	  float y0g = x0*sliceInfo.sinAlpha + y0*sliceInfo.cosAlpha;
+	  float z0g = z0;
+
+	  float x1g = x1*sliceInfo.cosAlpha - y1*sliceInfo.sinAlpha;
+	  float y1g = x1*sliceInfo.sinAlpha + y1*sliceInfo.cosAlpha;	  
+	  float z1g = z1;
+	  
+	  //cout<<x0<<" "<<y0<<" "<<z0<<" "<<x0g<<" "<<y0g<<" "<<z0g<<endl;
+	  //cout<<x1<<" "<<y1<<" "<<z1<<" "<<x1g<<" "<<y1g<<" "<<z1g<<endl;
+
+	  double xyz[3] = {x0g,y0g,z0g};
+	  double d[3] = {0,0,0};
+	  spaceChargeCorrection( xyz, d );
```

Comment:
I'm a bit puzzled by the fact that the correction function is called directly to calculate the parameters for the histogram. Shouldn't that be done in the transformation object?

Reply:
I have put comments there. 
I compare the original correction to the difference between (transform with correction - transform without correction). It should be the same within the roundoff errors.